folly/Random.cpp

```diff
@@ -26,16 +26,33 @@
 #include <folly/File.h>
 #include <folly/FileUtil.h>
 
+#ifdef _MSC_VER
+#include <wincrypt.h>
+#endif
+
 namespace folly {
 
 namespace {
 
 void readRandomDevice(void* data, size_t size) {
+#ifdef _MSC_VER
+  static bool initialized = false;
```

Comment:
That was actually deliberate, and is a result of my own personal form of insanity. As the impact of a second call to `CryptoAcquireContext` is minimal on resource usage, basically just incrementing a reference count, and as `cryptoProv` will be entirely written from a single instruction, there is no possibility of getting part of one thread's info and part of another's. It's perfectly fine for a second thread to be trying to initialize the provider at the same time, that's why it's setting initialized after the provider has been acquired.

If it's preferred to use a mutex though, I can certainly do that?